app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadViewModel.kt

```diff
@@ -29,22 +32,32 @@ import com.infomaniak.mail.data.models.thread.Thread
 import com.infomaniak.mail.ui.MainViewModel
 import io.realm.kotlin.MutableRealm
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.mapNotNull
 import kotlinx.coroutines.launch
 
 class ThreadViewModel : ViewModel() {
 
-    private val currentThread = MutableLiveData<Thread>()
-    val messages = Transformations.switchMap(currentThread) { thread ->
-        liveData(Dispatchers.IO) { emitSource(thread.messages.asFlow().asLiveData()) }
+    private var _thread: Thread? = null
+    val thread: Thread? get() = _thread
```

Comment:
Why ?

Reply:
So the Thread can't be reassigned outside of ThreadViewModel.

Reply:
It's been updated.